source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,807 @@
+    enet_tx_bd_struct_t * TxBuffDescrip;
+    rx_buffer_t * RxDataBuff;
+    tx_buffer_t * TxDataBuff;
+    rx_pbuf_wrapper_t RxPbufs[ ENET_RXBUFF_NUM ];
+};
+
+typedef enum xEMAC_STATE
+{
+    xEMAC_SetupPHY,
+    xEMAC_WaitPHY,
+    xEMAC_Init,
+    xEMAC_Ready,
+    xEMAC_Fatal,
+} EMACState_t;
+
+static EMACState_t eEMACState = xEMAC_SetupPHY;
+
+static mdio_handle_t mdioHandle = { .ops = &EXAMPLE_MDIO_OPS };
+
+static phy_handle_t phyHandle = { .phyAddr = 0x00, .mdioHandle = &mdioHandle, .ops = &EXAMPLE_PHY_OPS };
+
+/*
+ * A deferred interrupt handler task that processes
+ */
+static TaskHandle_t receiveTaskHandle = NULL;
+
+static struct ethernetif EthernetInterface1;
+struct ethernetif * ethernetifLocal = &EthernetInterface1;
+
+static bool bGlobalLinkStatus = false;
+
+/*-----------------------------------------------------------*/
+
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_rx_bd_struct_t rxBuffDescrip_0[ ENET_RXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_tx_bd_struct_t txBuffDescrip_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static rx_buffer_t rxDataBuff_0[ ENET_RXBUFF_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static tx_buffer_t txDataBuff_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+
+static void prvEMACHandlerTask( void * pvParameters );
+
+static void ethernet_callback( ENET_Type * base,
+                               enet_handle_t * handle,
+                               enet_event_t event,
+                               enet_frame_info_t * frameInfo,
+                               void * userData );
+
+static void prvProcessFrame( int length );
+
+static status_t xSetupPHY( phy_config_t * pxConfig,
+                           EMACState_t * peEMACState );
+
+static status_t xWaitPHY( phy_config_t * pxConfig,
+                          EMACState_t * peEMACState );
+
+static status_t xEMACInit( EMACState_t * peEMACState,
+                           phy_speed_t speed,
+                           phy_duplex_t duplex );
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+    status_t xStatus;
+    phy_config_t xConfig = { 0 };
+    BaseType_t xResult = pdFAIL;
+    phy_speed_t speed;
+    phy_duplex_t duplex;
+    BaseType_t xTaskCreated;
+    static BaseType_t xFirstCall = pdTRUE;
+
+    configASSERT( FSL_FEATURE_ENET_QUEUE == 1 );
+
+    #if defined( ENET_ENHANCEDBUFFERDESCRIPTOR_MODE )
+        /* This driver is not to be used with enhanced buffer descriptor mode. */
+        configASSERT( pdFALSE == pdTRUE );
+    #endif
+
+    switch( eEMACState )
+    {
+        case xEMAC_SetupPHY:
+            xStatus = xSetupPHY( &xConfig, &eEMACState );
+
+            if( xStatus != kStatus_Success )
+            {
+                eEMACState = xEMAC_Fatal;
+                break;
+            }
+
+        /* Fall through. */
+        case xEMAC_WaitPHY:
+            FreeRTOS_printf( ( "Configuration successful. Waiting for auto-negotiation to complete.\r\n" ) );
```

Comment:
I like these changes, it is very easy to understand the main logic.
Let's do some tests with DHCP enabled and unplug and plug the DUT.

Reply:
I tried and they succeeded. I have attached the log in the PR description.